rest-service/manager_rest/test/endpoints/test_deployment_labels_dependencies.py

```diff
@@ -0,0 +1,334 @@
+                }
+            ]
+        )
+
+        self.client.deployments.update_labels('dep_2', [
+                {
+                    'csys-obj-parent': 'dep_3'
+                }
+            ]
+        )
+
+        self.client.deployments.update_labels('dep_4', [
+                {
+                    'csys-obj-parent': 'dep_5'
+                }
+            ]
+        )
+
+    @patch('manager_rest.resource_manager.ResourceManager'
+           '.handle_deployment_labels_graph')
+    @patch('manager_rest.resource_manager.ResourceManager'
+           '.verify_deployment_parent_labels')
+    def test_deployment_with_empty_labels(self,
+                                          verify_parents_mock,
+                                          handle_labels_graph_mock):
+        self.put_deployment('deployment_with_no_labels')
+        verify_parents_mock.assert_not_called()
+        handle_labels_graph_mock.assert_not_called()
+
+    @patch('manager_rest.resource_manager.ResourceManager'
+           '.handle_deployment_labels_graph')
+    @patch('manager_rest.resource_manager.ResourceManager'
+           '.verify_deployment_parent_labels')
+    def test_deployment_with_non_parent_labels(self,
+                                               verify_parents_mock,
+                                               handle_labels_graph_mock):
+        self.put_deployment_with_labels([{'env': 'aws'}, {'arch': 'k8s'}])
+        verify_parents_mock.assert_not_called()
+        handle_labels_graph_mock.assert_not_called()
+
+    def test_deployment_with_single_parent_label(self):
+        self.put_deployment('parent')
+        self.put_deployment_with_labels([{'csys-obj-parent': 'parent'}])
+
+        # deployment response
+        deployment = self.client.deployments.get('parent')
+        self.assertEqual(deployment.sub_services_count, 1)
+        self.assertEqual(deployment.sub_environments_count, 0)
+
+    def test_deployment_with_multiple_parent_labels(self):
+        self.put_deployment(deployment_id='parent_1',
+                            blueprint_id='blueprint_1')
+        self.put_deployment(deployment_id='parent_2',
+                            blueprint_id='blueprint_2')
+        self.put_deployment_with_labels(
+            [
+                {
+                    'csys-obj-parent': 'parent_1'
+                },
+                {
+                    'csys-obj-parent': 'parent_2'
+                }
+            ]
+        )
+        deployment_1 = self.client.deployments.get('parent_1')
+        deployment_2 = self.client.deployments.get('parent_2')
+        self.assertEqual(deployment_1.sub_services_count, 1)
+        self.assertEqual(deployment_1.sub_environments_count, 0)
+        self.assertEqual(deployment_2.sub_services_count, 1)
+        self.assertEqual(deployment_2.sub_environments_count, 0)
+
+    def test_deployment_with_invalid_parent_label(self):
+        error_message = 'label `csys-obj-parent` that do not exist'
```

Comment:
This error message should say something like "does not exist" rather than "do not exist"- did I miss it in this PR or is it elsewhere?

Reply:
I think you miss it but will update it no worries